### PR TITLE
add method to get elbow positions and velocity and fixed timeout errors for no reliable real time

### PR DESCRIPTION
**Pull request by mustaffxx**

### CODE REVIEW

File: urx/urrobot.py [hunk 1]
```diff
@@ -87,6 +87,15 @@ def send_program(self, prog):
...
         self.secmon.send_program(prog)
 
+    def send_custom_program(self, prog):
+        """
+        send a complete program using urscript to the robot
+        the program is executed immediatly and any runnning
+        program is interrupted
+        """
+        self.logger.info("Sending program: " + prog)
+        self.rtmon.send_program(prog)
```
Comment (oroulet) [inline urx/urrobot.py:97]: what is the difference between that method and the exisitng one? it looks like it is using another interface. Why?
give it an explicit name and write in doc string what is the difference from the other one

File: urx/ursecmon.py [hunk 2]
```diff
@@ -340,7 +340,7 @@ def _get_data(self):
                 tmp = self._s_secondary.recv(1024)
                 self._dataqueue += tmp
 
-    def wait(self, timeout=0.5):
+    def wait(self, timeout=2.0):
```
Comment (oroulet) [inline urx/ursecmon.py:343]: I see you increased everywhere the timeouts. These timeouts were not random. if it takes you more than 0.5s to create a connection then you have an issue in your network

File: urx/urrtmon.py [hunk 3]
```diff
@@ -254,7 +283,7 @@ def __recv_rt_data(self):
             self._ctrlTimestamp = np.array(unp[0])
             if self._last_ctrl_ts != 0 and (
                     self._ctrlTimestamp -
-                    self._last_ctrl_ts) > 0.010:
+                    self._last_ctrl_ts) > 0.1:
```
Comment (oroulet) [inline urx/urrtmon.py:286]: same here. The robot sends data at 100Hz, the timestamp should be accordingly, if no packet is send in over 0.1s there is something bad happenign and we need to discover it

File: examples/get_realtime_data.py [hunk 4]
```diff
@@ -2,9 +2,15 @@
...
-r = urx.Robot("192.168.111.134", use_rt=True, urFirm=5.1)
 
 if __name__ == "__main__":
+    while 1:
+        try:
+            r = urx.Robot("192.168.137.129", use_rt=True, urFirm=5.1)
+            break
+        except:
+            pass
+
```
Comment (oroulet) [inline examples/get_realtime_data.py:13]: example code must be as simple as possible the lines over are useless complexity.
Looks like you have a network issue. That is the one that should be fixed, not every library on earth

File: examples/get_realtime_data.py [hunk 5]
```diff
@@ -14,6 +20,8 @@
             main_voltage = r.get_main_voltage()
             robot_voltage = r.get_robot_voltage()
             robot_current = r.get_robot_current()
+            elbow_position = r.get_elbow_position()
+            elbow_velocity = r.get_elbow_velocity()
```
Comment (oroulet) [inline examples/get_realtime_data.py:24]: this is a usefull adddition and is fine

File: urx/ursecmon.py (hunk 2 above)
Comment (mustaffxx) [reply urx/ursecmon.py:343]: examining the issues and forks, I could see that several people had these timeout errors when creating the connection, which were easily "resolved" by increasing the timeout to become usable

**Comment by mustaffxx** [comment]
this PR is really poor and i will submit new one only with elbow methods and timeout incresead

**Pull request closed without merging**